File: 백준/Silver/1764. 듣보잡/듣보잡.py
import sys

# n : 듣도 보도 못한 사람의 수
# m : 보도 못한 사람의 수

n, m = map(int, sys.stdin.readline().split())

bd = {}
db = {}

for i in range(n):
    bd[sys.stdin.readline().strip()] = 1
answer = 0
names = []
for i in range(m):
    name = sys.stdin.readline().strip()
    if bd.get(name):
        answer += 1
        names.append(name)

names.sort()
print(answer)
for i in names:
    print(i)

File: 백준/Silver/1764. 듣보잡/듣보잡.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        HashSet<String> dm = new HashSet<>();

        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        for (int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            dm.add(st.nextToken());
        }
        ArrayList<String> answer = new ArrayList<>();
        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());
            String name = st.nextToken();
            if (dm.contains(name)) {
                answer.add(name);
            }
        }
        Collections.sort(answer);

        StringBuilder sb = new StringBuilder();
        sb.append(answer.size()).append("\n");
        for (String s : answer) {
            sb.append(s).append("\n");
        }
        System.out.println(sb);


    }
}